pkg/config/config.go

```diff
@@ -981,6 +981,21 @@ func getMultipleEndpointsWithConfig(config Config) (map[string][]string, error)
 	return keysPerDomain, nil
 }
 
+// ComputeAPIDomain returns a domain to be used for API validation
+func ComputeAPIDomain(domain string) string {
+	apiDomain := ""
+	if parsedDomain, err := url.Parse(domain); err == nil {
```

Comment:
just a style nit: it's generally easier to read the code when it has fewer nested `if` statements, and in general in Go we want to eliminate error cases as they occur and return in those cases (see https://golang.org/doc/effective_go.html#if).

So instead I'd do something like:

```go
parsedDomain, err := url.Parse(domain)
if err != nil {
  return "", fmt.Errorf("could not parse '%s' as a URL", domain)
}

if _, found := ddURLs[parsedDomain.Host]; found {
  [...]
```